mlir/include/mlir/Dialect/ArmNeon/ArmNeon.td

```diff
@@ -30,6 +30,15 @@ def ArmNeon_Dialect : Dialect {
   // to the LLVMDialect (ops or types).
 }
 
+//===----------------------------------------------------------------------===//
+// ArmNeon type definition
+//===----------------------------------------------------------------------===//
+
+class NeonVectorOfLength<int length, Type elementType> : ShapedContainerType<
+  [elementType], And<[IsVectorOfShape<[length]>, IsFixedVectorTypePred]>,
+  "Neon vector of " # length # "x" # elementType.summary,
```

Comment:
Sorry, I didn't look how the summary was combined so it's a little wrong now (repeats the type twice)

I think this would work better:
```suggestion
  "a vector with length " # length"
```

Should go from:
```
op operand #1 must be Neon vector of 16x8-bit signless integer of 8-bit signless integer values
```
to:
```
op operand #1 must be a vector with length 16 of 8-bit signless integer values
```